lib/sycamore/sycamore/transforms/standardizer.py

```diff
@@ -0,0 +1,162 @@
+from dateutil import parser
+import re
+from typing import List
+
+
+class Standardizer(ABC):
+    """
+    An Abstrant class for implementing Standarizers.
+    """
+
+    @abstractmethod
+    def standardize(self, doc: Document, key_path: List[str]) -> Document:
+        pass
+
+
+class LocationStandardizer(Standardizer):
+    """
+    This Class standardizes the format of US State abbreviations.
+    """
+
+    def __init__(self):
+        self._state_abbreviations = {
+            "AK": "Alaska",
+            "AL": "Alabama",
+            "AR": "Arkansas",
+            "AZ": "Arizona",
+            "CA": "California",
+            "CO": "Colorado",
+            "CT": "Connecticut",
+            "DC": "District of Columbia",
+            "DE": "Delaware",
+            "FL": "Florida",
+            "GA": "Georgia",
+            "HI": "Hawaii",
+            "IA": "Iowa",
+            "ID": "Idaho",
+            "IL": "Illinois",
+            "IN": "Indiana",
+            "KS": "Kansas",
+            "KY": "Kentucky",
+            "LA": "Louisiana",
+            "MA": "Massachusetts",
+            "MD": "Maryland",
+            "ME": "Maine",
+            "MI": "Michigan",
+            "MN": "Minnesota",
+            "MO": "Missouri",
+            "MS": "Mississippi",
+            "MT": "Montana",
+            "NC": "North Carolina",
+            "ND": "North Dakota",
+            "NE": "Nebraska",
+            "NH": "New Hampshire",
+            "NJ": "New Jersey",
+            "NM": "New Mexico",
+            "NV": "Nevada",
+            "NY": "New York",
+            "OH": "Ohio",
+            "OK": "Oklahoma",
+            "OR": "Oregon",
+            "PA": "Pennsylvania",
+            "RI": "Rhode Island",
+            "SC": "South Carolina",
+            "SD": "South Dakota",
+            "TN": "Tennessee",
+            "TX": "Texas",
+            "UT": "Utah",
+            "VA": "Virginia",
+            "VT": "Vermont",
+            "WA": "Washington",
+            "WI": "Wisconsin",
+            "WV": "West Virginia",
+            "WY": "Wyoming",
+        }
+
+    def replace_abbreviations(self, input_string):
+        """
+        This method replaces the US State abbreviations with full names.
+        """
+        state_abbreviations = {key: value for key, value in self._state_abbreviations.items()}
+        abbreviations = {**state_abbreviations}
+
+        pattern = re.compile(r"\b(" + "|".join(re.escape(key) for key in abbreviations.keys()) + r")\b")
+
+        def replace_match(match):
+            return abbreviations[match.group(0)]
+
+        result_string = pattern.sub(replace_match, input_string)
+        return result_string
+
+    def standardize(self, doc: Document, key_path: List[str]) -> Document:
+        """
+        This Methods creates a new element standardises the location property of the Element
+        by replcaing all instance of US State abbreviation with correct name.
+        """
+        current = doc
```

Comment:
This block of code is duplicated across both standardizers. Should this be part of a parent class or something?